crates/faucet/src/handlers.rs

```diff
@@ -46,45 +49,53 @@ pub async fn get_tokens(
     let asset =
         FungibleAsset::new(state.id, state.asset_amount).expect("Failed to instantiate asset.");
 
+    // Instantiate note type
+    let note_type = NoteType::OffChain;
+
     // Instantiate transaction template
-    let tx_template = TransactionTemplate::MintFungibleAsset { asset, target_account_id };
+    let tx_template = TransactionTemplate::MintFungibleAsset(asset, target_account_id, note_type);
+
+    // Instantiate transaction request
+    let tx_request = client
+        .lock()
+        .await
+        .build_transaction_request(tx_template)
+        .expect("Failed to build transaction request.");
 
     // Run transaction executor & execute transaction
     let tx_result = client
         .lock()
         .await
-        .new_transaction(tx_template)
+        .new_transaction(tx_request)
         .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
 
-    // Get note id
-    let note_id: NoteId = tx_result
-        .created_notes()
-        .first()
-        .ok_or_else(|| {
-            FaucetError::InternalServerError("Failed to access generated note.".to_string())
-        })?
-        .id();
+    // Get created notes from transaction result
+    let created_notes = tx_result.created_notes().clone();
 
     // Run transaction prover & send transaction to node
-    {
-        let mut client_guard = client.lock().await;
-        client_guard
-            .send_transaction(tx_result)
-            .await
-            .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
-    }
-
-    // Get note from client store
-    let input_note = state
-        .client
-        .clone()
+    client
         .lock()
         .await
-        .get_input_note(note_id)
+        .submit_transaction(tx_result)
+        .await
         .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
 
-    // Serialize note for transport
-    let bytes = input_note.to_bytes();
+    let note_id: NoteId;
+
+    // Serialize note into bytes
+    let bytes = match created_notes.first() {
+        Some(note) => {
+            note_id = note.id();
+            note.to_bytes()
+        },
```

Comment:
Have you tested this on the client? I think in order to import this you might need an `InputNoteRecord`.

Reply:
Indeed, fixed.